internal/db/repos/instance.go

```diff
@@ -68,6 +68,34 @@ func (r *InstanceRepository) UpdateByID(ctx context.Context, id uint, instance *
 	})
 }
 
+// UpdateIPByName updates the public IP of an instance by its name
+func (r *InstanceRepository) UpdateIPByName(ctx context.Context, name string, ip string) error {
+	result := r.db.WithContext(ctx).Model(&models.Instance{}).
+		Where("name = ? AND deleted_at IS NULL", name).
+		Update("public_ip", ip)
+	if result.Error != nil {
+		return fmt.Errorf("failed to update instance IP: %w", result.Error)
+	}
+	if result.RowsAffected == 0 {
+		return fmt.Errorf("no instance found with name: %s", name)
+	}
+	return nil
+}
+
+// UpdateStatus updates the status of an instance
+func (r *InstanceRepository) UpdateStatus(ctx context.Context, ID uint, status models.InstanceStatus) error {
+	return r.db.WithContext(ctx).Model(&models.Instance{}).
+		Where(&models.Instance{Model: gorm.Model{ID: ID}}).
+		Update("status", status).Error
+}
+
+// UpdateStatusByName updates the status of an instance by its name
+func (r *InstanceRepository) UpdateStatusByName(ctx context.Context, name string, status models.InstanceStatus) error {
+	return r.db.WithContext(ctx).Model(&models.Instance{}).
```

Comment:
I think these are conflicts. 
We don't need all these `Update<field>` methods because `gorm` will only update non-null fields. 
Main should be updated with the minimum required `UpdateByID` and `UpdateByName`